### PR TITLE
Navbar implementation

### LINKED ISSUE #2: Navbar

### PR DESCRIPTION
**Pull request by parth4apple**
### Administrative Info
[GitHub Issue 2]

### Changes
Added styled-components to package.json for component styling
Created Navbar Component and its styling based on the figma model
Used the Navbar Component in the jsx of Landing.js, Foster.js, Dashboard.js, and Calendar.js files 

### Testing
The changes I made look and behave like the figma model

### Confirmation of Change 
navbar

https://user-images.githubusercontent.com/72187062/149869581-95c25485-9d6c-445e-95bf-cc21d7b60165.mov

Edit: I implemented the mobile view for the navbar and made it responsive to different viewports. A video demonstration of how it works is shown below.

https://user-images.githubusercontent.com/72187062/151734194-0cfde936-dac0-422b-925b-4fbecfc2dec7.mov

### CODE REVIEW

**Comment by portoaj** [comment]
Can you remove your changes to all the pages, and just push the Navbar/ Navbar Styled components. One issue with this commit is that it is going to conflict with my updates to the dashboard page.

**Review by dnanjunda — CHANGES_REQUESTED** [review]
Nice job Parth! This looks good overall but there's a couple things to fix:

1. I think the web navbar here is a bit too wide. Make sure the width matches whatever is specified on Figma. 
navbar

2. Right now the text doesn't look great when window width is small. Use vw/vh for font size so that it is responsive to resizing. You can use on online converter to convert px to vw/vh.
navbar_font

3. For some of the mobile widths, the navbar's position is a bit off. Make sure it looks good for all the different devices in inspect element.
navbar_mobile

4. Can you add a "Sign Out" button at the bottom of the navbar for mobile, like in the designs?
5. Right now you have a .js file for your navbar css which is fine, but just to maintain consistency with the rest of the project and TSE in general, make a Navbar.css file in the css folder and put all the navbar related css in there.

**Review by dnanjunda — COMMENTED** [review]
Almost there Parth! Just a couple small things:

1. I don't think you moved the css code for the navbar from your .js file to a .css file in the css folder. Please do this.
2. This is a minor thing, but can you add a slide out animation for the mobile navbar when it is open and the hamburger icon is clicked, similar to the slide in you already have? If it's hard to do, don't worry about it.

File: client/package.json [hunk 1]
```diff
@@ -6,10 +6,12 @@
     "@testing-library/jest-dom": "^5.16.1",
     "@testing-library/react": "^12.1.2",
     "@testing-library/user-event": "^13.5.0",
+    "package.json": "^2.0.1",
```
Comment (petabite) [inline client/package.json:9]:
```suggestion
```

Comment (petabite) [inline client/package.json]:
```suggestion
```

File: client/src/components/Navbar.js [hunk 2]
```diff
@@ -1,10 +1,183 @@
...
+    top: 15%;
+    right: 0%;
+    width: min(300px, 45vw);
+    z-index: 5;
+
+    animation-name: ${slideIn};
+    animation-duration: 0.5s;
+    animation-iteration-count: 1;
+    }
+  }
```
Comment (petabite) [inline client/src/components/Navbar.js]: extra curly bracket?

File: client/src/components/Navbar.js [hunk 3]
```diff
@@ -1,10 +1,183 @@
...
+
+    background: none;
+    color: inherit;
+    border: none;
+    padding: 0;
+    font: inherit;
+    cursor: pointer;
+    outline: inherit;
+
+}
```
Comment (petabite) [inline client/src/components/Navbar.js]: here too

File: package.json [hunk 4]
```diff
@@ -16,7 +16,8 @@
   "dependencies": {
     "dotenv": "^11.0.0",
     "express": "^4.17.2",
-    "mongoose": "^6.1.6"
+    "mongoose": "^6.1.6",
+    "styled-components": "^5.3.3"
```
Comment (petabite) [inline package.json]:
```suggestion
```
I think you may have accidentally installed `styled-components` in the root folder. Can you run `npm uninstall` on this? This should also resolve the merge conflict you have.

**Review by petabite — CHANGES_REQUESTED** [review]
Nice, looking good! Everything is responsive on my end and is working as expected. Left a couple of comments for some small things below. Should be quick fixes.

**Review by dnanjunda — APPROVED** [review]
LGTM. Great job Parth!

**Review by petabite — APPROVED** [review]
👌

**Pull request merged**